src/overlays/actors/ovl_Obj_Aqua/z_obj_aqua.c

```diff
@@ -26,55 +35,266 @@ const ActorInit Obj_Aqua_InitVars = {
+    s32 i;
 
-extern ColliderCylinderInit D_80ACC2C0;
-extern InitChainEntry D_80ACC2EC[];
+    effectPos.y = this->actor.floorHeight;
+    for (i = 0; i < 4; i++) {
+        effectPos.x = (this->actor.world.pos.x + Math_SinS((s32)(Rand_ZeroOne() * 7200.0f) + angleOffset) * 8.0f);
+        effectPos.z = (this->actor.world.pos.z + Math_CosS((s32)(Rand_ZeroOne() * 7200.0f) + angleOffset) * 8.0f);
+        EffectSsGSplash_Spawn(globalCtx, &effectPos, NULL, NULL, 0, 120);
+        angleOffset += 0x4000;
+    }
+    effectPos.x = this->actor.world.pos.x;
+    effectPos.z = this->actor.world.pos.z;
+    EffectSsGSplash_Spawn(globalCtx, &effectPos, NULL, NULL, 0, 300);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Aqua/func_80ACB6A0.s")
+void func_80ACB940(ObjAqua* this, GlobalContext* globalCtx) {
+    s32 pad;
+    Vec3f effectPos;
+    Vec3f effectVel;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Aqua/func_80ACB7F4.s")
+    effectVel.x = Rand_ZeroOne() - 0.5f;
+    effectVel.y = 2.0f;
+    effectVel.z = Rand_ZeroOne() - 0.5f;
+    effectPos.x = this->actor.world.pos.x + (effectVel.x * 40.0f);
+    effectPos.y = this->actor.world.pos.y;
+    effectPos.z = this->actor.world.pos.z + (effectVel.z * 40.0f);
+    EffectSsIceSmoke_Spawn(globalCtx, &effectPos, &effectVel, &D_801D15B0, (s32)(Rand_ZeroOne() * 24.0f) + 70);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Aqua/func_80ACB940.s")
+void func_80ACBA10(ObjAqua* this) {
+    s32 pad;
+    MtxF sp2C;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Aqua/func_80ACBA10.s")
+    func_800C0094(this->actor.floorPoly, this->actor.world.pos.x, this->actor.floorHeight, this->actor.world.pos.z,
+                  &sp2C);
+    func_8018219C(&sp2C, &this->actor.shape.rot, 0);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Aqua/func_80ACBA60.s")
+s32 func_80ACBA60(ObjAqua* this, GlobalContext* globalCtx) {
+    s32 pad;
+    WaterBox* waterBox;
+    f32 ySurface;
+    s32 bgId;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Aqua/ObjAqua_Init.s")
+    if (func_800C9EBC(globalCtx, &globalCtx->colCtx, this->actor.world.pos.x, this->actor.world.pos.z, &ySurface,
+                      &waterBox, &bgId) &&
+        (this->actor.world.pos.y < ySurface)) {
+        return 1;
+    }
+    return 0;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Aqua/ObjAqua_Destroy.s")
+void ObjAqua_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ObjAqua* this = THIS;
+    s32 i;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Aqua/func_80ACBC70.s")
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    this->actor.scale.x = 0.0009f;
+    this->actor.scale.y = 0.0005f;
+    this->actor.scale.z = 0.0009f;
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    ActorShape_Init(&this->actor.shape, 0.0f, func_800B3FC0, 60.0f);
+    if (0) {};
```

Comment:
```suggestion
    if (1) {};
```
If it still matches, this would be preferred